cleanup: move the test of TaintBasedEvictions features to sig-node

fengzixu opened the pull request:
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

1. move the integration test of TaintBasedEvictions to test/integration/node
2. move the e2e test of TaintBasedEvictions to e2e_node

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #87427

**Special notes for your reviewer**:
```release-note
NONE
```

fengzixu commented:
@Huang-Wei @damemi 

Hi guys. I finished the moving tests from sig-scheduling to sig-node based on #87427.

Can you help me review it and ping members of sig-node. Thank you so much.

damemi commented:
/ok-to-test

damemi commented:
cc @kubernetes/sig-node-pr-reviews

mattjmcnaughton commented on test/e2e_node/BUILD:
```diff
@@ -135,6 +135,7 @@ go_test(
         "startup_probe_test.go",
         "summary_test.go",
         "system_node_critical_test.go",
+        "taints.go"
```
I think you're missing a comma here :) which is I think responsible for pretty much all of the test failures :)

damemi replied:
yeah, not sure if this was done manually but if so running `./hack/update-bazel.sh` is how these should be updated

fengzixu replied:
Is BUILD file can be generated automatically? Any script can do it?

mattjmcnaughton replied:
Yup, `hack/update-bazel.sh` is the script which will do it.

mattjmcnaughton commented on test/integration/node/OWNERS:
```diff
@@ -0,0 +1,5 @@
+# See the OWNERS docs at https://go.k8s.io/owners
+
+approvers:
```
Nit: Why are we not specifying approvers?

fengzixu replied:
emm. Actually, I don't know who I should specify. sig-node or sig-scheduling. Can you give me hint?

fengzixu replied:
Let me fill it.

fengzixu replied:
Who I need to speficy? It seems like sig-node-maintainers is incorrect.

mattjmcnaughton replied:
`sig-node-approvers` is the proper alias.

mattjmcnaughton commented on test/integration/node/util.go:
```diff
@@ -0,0 +1,350 @@
+package node
```
Perhaps I'm missing something, but where is this `util.go` code coming from? I don't see the equivalent code removed anywhere. Are we duplicating it?

fengzixu replied:
Actually, some functions which are used by the integration test of TaintBasedEvictions exists in the `test/integration/scheduling/util.go`. In my opinion, if we move the TaintBasedEviction test to the sig-node, we shouldn't import unnecessary functions from sig-scheduling.

mattjmcnaughton replied:
Hmm, I definitely agree that we don't want to import from `test/integration/scheduling/util.go`. However, I'd prefer to avoid the duplication if we can. Can we move `util.go` to a more general package which both `scheduling` and `node` could safetly import from?

Alternatively, if there's a strict timeline for getting this merged, we could also address this duplication in a later pr. If we go that route, could you please add a TODO to resolve the duplication?

Huang-Wei replied:
`test/integration/util` would be a good place.

fengzixu replied:
Yep. Let me move it .

mattjmcnaughton reviewed (commented):
Thanks for this pr :)

Agree with the high-level goal. I have a couple of impl qs :)

fengzixu commented:
/remove sig/api-machinery
/remove sig/apps
/remove sig/autoscaling
/remove sig/cli

neolit123 commented:
/remove-area kubeadm

Huang-Wei commented on test/e2e_node/taints.go:
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 */
 
-package scheduling
+package e2enode
```
`test/e2e_node` uses a totally different e2e mechanism. We should move it to `test/e2e/node`.

Huang-Wei commented on test/integration/node/OWNERS:
```diff
@@ -0,0 +1,6 @@
+# See the OWNERS docs at https://go.k8s.io/owners
+
+approvers:
+- sig-node-maintainers
```
```suggestion
- sig-node-approvers
```

Huang-Wei reviewed (commented):
Some comments.

fengzixu commented:
@manansaraf @Huang-Wei 

Hey guys. I revised this PR according to the comments above.  Please review it again.

mattjmcnaughton reviewed (commented):
Mind addressing the failures? https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/88152/pull-kubernetes-verify/1229445261944688641

Lmk if there any questions around how to address the failures - you can verify if they are working via the `hack/verify-*` scripts.

Happy to review if you have specific questions or once the test suite is green. Thanks!

fengzixu commented:
> Mind addressing the failures? https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/88152/pull-kubernetes-verify/1229445261944688641
> 
> Lmk if there any questions around how to address the failures - you can verify if they are working via the `hack/verify-*` scripts.
> 
> Happy to review if you have specific questions or once the test suite is green. Thanks!

I see. I missed it. Let me revise it.

fengzixu commented:
@mattjmcnaughton Not sure why the pull-kubernetes-bazel-test  always failed. 
The report of the test doesn't give accurate info: https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/88152/pull-kubernetes-bazel-test/1229637653826113536

fengzixu commented:
/retest

fengzixu commented:
@Huang-Wei @mattjmcnaughton It seems like all tests have passed. So please review it

fedebongio commented:
/remove-sig api-machinery

mattjmcnaughton reviewed (approved):
/lgtm

Thanks for this change and for addressing all the review feedback.

The one thing I'm not sure about, and would like someone to sign-off on, is whether we need to make any other changes similar to the change to the `conformance.txt` test file.

Huang-Wei commented on test/integration/node/main_test.go:
```diff
@@ -0,0 +1,27 @@
+/*
+Copyright 2017 The Kubernetes Authors.
```
nit: 2020

fengzixu replied:
fixed

Huang-Wei commented:
/approve
for the scheduling side. Could you squash the commits?

fengzixu commented:
@Huang-Wei @smarterclayton @mattjmcnaughton
I have rebased upstream and combine multiple commits to one. Please review it.

/assign @smarterclayton

mattjmcnaughton reviewed (approved):
/lgtm

Still good on my end if all you did was squash (again, modulo open conformance question).

Will let @smarterclayton give final sign-off.

fengzixu commented:
conflict happened on /test/e2e/node/BUILD file. So I rebase it.

fengzixu commented:
Ping @smarterclayton :)

damemi reviewed (approved):
/lgtm
will need an approve though

smarterclayton commented:
All refactoring of utility methods should be split into its own PR and merged first.  This PR must be two commits - one changing the conformance file, and one moving the files and renaming things.

smarterclayton commented:
/hold

fengzixu commented:
> All refactoring of utility methods should be split into its own PR and merged first. This PR must be two commits - one changing the conformance file, and one moving the files and renaming things.

has split, please review. @smarterclayton

mattjmcnaughton commented:
> > All refactoring of utility methods should be split into its own PR and merged first. This PR must be two commits - one changing the conformance file, and one moving the files and renaming things.
> 
> has split, please review. @smarterclayton

@smarterclayton can confirm, but I think that he was actually suggesting two separate PRs, not one PR containing two commits (which is what I believe we have now). I think we want the ability to first merge the utility methods and then merge the conformance changes.

Do you mind creating two separate PRs so that we can merge the changes independently? If you tag me in both, happy to review tomorrow morning. Thanks :)

fengzixu commented:
> > > All refactoring of utility methods should be split into its own PR and merged first. This PR must be two commits - one changing the conformance file, and one moving the files and renaming things.
> > 
> > 
> > has split, please review. @smarterclayton
> 
> @smarterclayton can confirm, but I think that he was actually suggesting two separate PRs, not one PR containing two commits (which is what I believe we have now). I think we want the ability to first merge the utility methods and then merge the conformance changes.
> 
> Do you mind creating two separate PRs so that we can merge the changes independently? If you tag me in both, happy to review tomorrow morning. Thanks :)
> 
> Alternatively, if the two separate PRs already exist, and I'm just missing them, do you mind linking them?

@mattjmcnaughton Sure. Let me split it into 2 PRs.

fengzixu commented:
https://github.com/kubernetes/kubernetes/pull/88685 @smarterclayton @mattjmcnaughton Have split, PTAL

mattjmcnaughton commented:
I think we will need to rebase after https://github.com/kubernetes/kubernetes/pull/88685 is merged? Mind pinging me once that's done?

Thanks :)

damemi commented:
@fengzixu @smarterclayton with code freeze this week, will this be able to merge? The other 2 PRs to complete this enhancement (https://github.com/kubernetes/kubernetes/pull/87487 & https://github.com/kubernetes/website/pull/19302) are predicated on this merging first

fengzixu commented:
PR https://github.com/kubernetes/kubernetes/pull/88685 has been merged last night. I will rebase for this PR this night. After that, please @smarterclayton review and approve it.

@damemi Is it ok?

fengzixu committed:
cleanup: move the test of TaintBasedEvictions features to sig-node

1. move the integration test of TaintBasedEvictions to test/integration/node
2. move the e2e test of TaintBasedEvictions e2e test/e2e/node
3. modify the conformance file to adapt the TaintBasedEviction test

fengzixu commented:
/retest

fengzixu commented:
@mattjmcnaughton @smarterclayton  Hi, I have rebased it and all test cases seem passed. Please review this PR. Because of the code freeze will start at the end of this week.

mattjmcnaughton reviewed (commented):
Will let @smarterclayton give the final sign-off since I don't have approval permissions.

/assign @smarterclayton

damemi reviewed (approved):
/lgtm
Thanks for this @fengzixu!

damemi commented:
Bumping, @smarterclayton will you have time to review this today for code freeze? If not can we file an exception for the enhancement?

smarterclayton commented:
/approve

fengzixu commented:
/hold cancel
